game.py

```diff
@@ -20,7 +20,30 @@ def snowman(snowman_word):
     If the player wins and, 
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    wrong_guesses_list = []
+    wrong_guesses_count = 0
+    
+    while wrong_guesses_count < SNOWMAN_MAX_WRONG_GUESSES:
+        print_snowman_graphic(wrong_guesses_count)
+        print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+        print("Wrong guesses so far:", " ".join(wrong_guesses_list))
+        
+        guess = get_letter_from_user(correct_letter_guess_statuses, wrong_guesses_list)
+        
+        if guess in snowman_word:
+            correct_letter_guess_statuses[guess] = True
+        else:
+            wrong_guesses_list.append(guess)
+            wrong_guesses_count += 1
+        
+        if is_word_guessed(snowman_word, correct_letter_guess_statuses):
+            print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+            print("Congratulations, you win!")
+            return
```

Comment:
Using a return statement to end the loop as soon as a win condition is met is such a smart move! I am a huge advocate for ending a function early when it makes sense!